diff --git a/src/MessageHandler/DeleteImagePostHandler.php b/src/MessageHandler/DeleteImagePostHandler.php
new file mode 100644
index 0000000..346d3d8
--- /dev/null
+++ b/src/MessageHandler/DeleteImagePostHandler.php
@@ -0,0 +1,12 @@
+<?php
+
+namespace App\MessageHandler;
+
+use App\Message\DeleteImagePost;
+
+class DeleteImagePostHandler
+{
+    public function __invoke(DeleteImagePost $deleteImagePost)
+    {
+    }
+}
