diff --git a/src/MessageHandler/DeleteImagePostHandler.php b/src/MessageHandler/DeleteImagePostHandler.php
index 386d722..76fe458 100644
--- a/src/MessageHandler/DeleteImagePostHandler.php
+++ b/src/MessageHandler/DeleteImagePostHandler.php
@@ -3,14 +3,26 @@
 namespace App\MessageHandler;
 
 use App\Message\DeleteImagePost;
+use App\Photo\PhotoFileManager;
+use Doctrine\ORM\EntityManagerInterface;
 
 class DeleteImagePostHandler
 {
+    private $photoManager;
+    private $entityManager;
+
+    public function __construct(PhotoFileManager $photoManager, EntityManagerInterface $entityManager)
+    {
+        $this->photoManager = $photoManager;
+        $this->entityManager = $entityManager;
+    }
+
     public function __invoke(DeleteImagePost $deleteImagePost)
     {
-        $photoManager->deleteImage($imagePost->getFilename());
+        $imagePost = $deleteImagePost->getImagePost();
+        $this->photoManager->deleteImage($imagePost->getFilename());
 
-        $entityManager->remove($imagePost);
-        $entityManager->flush();
+        $this->entityManager->remove($imagePost);
+        $this->entityManager->flush();
     }
 }
