diff --git a/src/Controller/ImagePostController.php b/src/Controller/ImagePostController.php
index 6e4ca86..4600088 100644
--- a/src/Controller/ImagePostController.php
+++ b/src/Controller/ImagePostController.php
@@ -62,7 +62,7 @@ class ImagePostController extends AbstractController
 
         $message = new AddPonkaToImage($imagePost->getId());
         $envelope = new Envelope($message, [
-            //new DelayStamp(500)
+            new DelayStamp(10000)
         ]);
         $messageBus->dispatch($envelope);
 
