diff --git a/.env b/.env
index 6bac647..7286f3e 100644
--- a/.env
+++ b/.env
@@ -29,7 +29,7 @@ DATABASE_URL=mysql://root:@127.0.0.1:3306/messenger_tutorial
 
 ###> symfony/messenger ###
 # Choose one of the transports below
-# MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages
-MESSENGER_TRANSPORT_DSN=doctrine://default
+MESSENGER_TRANSPORT_DSN=amqp://owryicqx:ueHzuUh1B3DXQNV0jZq_POrknNA3fXEk@fish.rmq.cloudamqp.com/owryicqx
+# MESSENGER_TRANSPORT_DSN=doctrine://default
 # MESSENGER_TRANSPORT_DSN=redis://localhost:6379/messages
 ###< symfony/messenger ###
