diff --git a/src/Controller/ImagePostController.php b/src/Controller/ImagePostController.php
index 0433d13..d3aceee 100644
--- a/src/Controller/ImagePostController.php
+++ b/src/Controller/ImagePostController.php
@@ -16,6 +16,7 @@ use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Messenger\Envelope;
 use Symfony\Component\Messenger\MessageBusInterface;
 use Symfony\Component\Messenger\Stamp\DelayStamp;
+use Symfony\Component\Messenger\Transport\AmqpExt\AmqpStamp;
 use Symfony\Component\Routing\Annotation\Route;
 use Symfony\Component\Validator\Constraints\Image;
 use Symfony\Component\Validator\Constraints\NotBlank;
@@ -62,7 +63,10 @@ class ImagePostController extends AbstractController
 
         $message = new AddPonkaToImage($imagePost->getId());
         $envelope = new Envelope($message, [
-            new DelayStamp(1000)
+            new DelayStamp(1000),
+            // pretend we want to route this message via a different
+            // routing key than its transport uses by default
+            new AmqpStamp('normal')
         ]);
         $messageBus->dispatch($envelope);
 
