diff --git a/src/MessageHandler/DeleteImagePostHandler.php b/src/MessageHandler/DeleteImagePostHandler.php
index 264bdf9..27a5db0 100644
--- a/src/MessageHandler/DeleteImagePostHandler.php
+++ b/src/MessageHandler/DeleteImagePostHandler.php
@@ -3,27 +3,31 @@
 namespace App\MessageHandler;
 
 use App\Message\DeleteImagePost;
+use App\Message\DeletePhotoFile;
 use App\Photo\PhotoFileManager;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
 
 class DeleteImagePostHandler implements MessageHandlerInterface
 {
-    private $photoManager;
+    private $messageBus;
     private $entityManager;
 
-    public function __construct(PhotoFileManager $photoManager, EntityManagerInterface $entityManager)
+    public function __construct(MessageBusInterface $messageBus, EntityManagerInterface $entityManager)
     {
-        $this->photoManager = $photoManager;
+        $this->messageBus = $messageBus;
         $this->entityManager = $entityManager;
     }
 
     public function __invoke(DeleteImagePost $deleteImagePost)
     {
         $imagePost = $deleteImagePost->getImagePost();
-        $this->photoManager->deleteImage($imagePost->getFilename());
+        $filename = $imagePost->getFilename();
 
         $this->entityManager->remove($imagePost);
         $this->entityManager->flush();
+
+        $this->messageBus->dispatch(new DeletePhotoFile($filename));
     }
 }
