diff --git a/src/MessageHandler/Event/RemoveFileWhenImagePostDeleted.php b/src/MessageHandler/Event/RemoveFileWhenImagePostDeleted.php
index 4897557..c90f380 100644
--- a/src/MessageHandler/Event/RemoveFileWhenImagePostDeleted.php
+++ b/src/MessageHandler/Event/RemoveFileWhenImagePostDeleted.php
@@ -6,7 +6,7 @@ use App\Message\Event\ImagePostDeletedEvent;
 use App\Photo\PhotoFileManager;
 use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
 
-class RemoveFileWhenImagePostDeleted
+class RemoveFileWhenImagePostDeleted implements MessageHandlerInterface
 {
     private $photoFileManager;
 
