diff --git a/src/Messenger/ExternalJsonMessageSerializer.php b/src/Messenger/ExternalJsonMessageSerializer.php
index 6cf1eca..5796ceb 100644
--- a/src/Messenger/ExternalJsonMessageSerializer.php
+++ b/src/Messenger/ExternalJsonMessageSerializer.php
@@ -61,7 +61,7 @@ class ExternalJsonMessageSerializer implements SerializerInterface
         ];
     }
 
-    private function createLogEmojiEnvelope($data): Envelope
+    private function createLogEmojiEnvelope(array $data): Envelope
     {
         if (!isset($data['emoji'])) {
             throw new MessageDecodingFailedException('Missing the emoji key!');
