diff --git a/src/Controller/ImagePostController.php b/src/Controller/ImagePostController.php
index c516ea6..f9605d8 100644
--- a/src/Controller/ImagePostController.php
+++ b/src/Controller/ImagePostController.php
@@ -71,7 +71,7 @@ class ImagePostController extends AbstractController
         ]);
         $messageBus->dispatch($envelope);
 
-        $messageBus->dispatch(new LogEmoji(2));
+        //$messageBus->dispatch(new LogEmoji(2));
 
         return $this->toJson($imagePost, 201);
     }
