diff --git a/src/Messenger/ExternalJsonMessageSerializer.php b/src/Messenger/ExternalJsonMessageSerializer.php
new file mode 100644
index 0000000..406818b
--- /dev/null
+++ b/src/Messenger/ExternalJsonMessageSerializer.php
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\Messenger;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\MessageDecodingFailedException;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+class ExternalJsonMessageSerializer implements SerializerInterface
+{
+    public function decode(array $encodedEnvelope): Envelope
+    {
+        // TODO: Implement decode() method.
+    }
+
+    public function encode(Envelope $envelope): array
+    {
+        // TODO: Implement encode() method.
+    }
+}
