diff --git a/src/MessageHandler/AddPonkaToImageHandler.php b/src/MessageHandler/AddPonkaToImageHandler.php
index 23932b5..f010cbe 100644
--- a/src/MessageHandler/AddPonkaToImageHandler.php
+++ b/src/MessageHandler/AddPonkaToImageHandler.php
@@ -44,6 +44,10 @@ class AddPonkaToImageHandler implements MessageHandlerInterface, LoggerAwareInte
             return;
         }
 
+        if (rand(0, 10) < 7) {
+            throw new \Exception('I failed randomly!!!!');
+        }
+
         $updatedContents = $this->ponkaficator->ponkafy(
             $this->photoManager->read($imagePost->getFilename())
         );
