diff --git a/src/Controller/ImagePostController.php b/src/Controller/ImagePostController.php
index 4746370..bf609a9 100644
--- a/src/Controller/ImagePostController.php
+++ b/src/Controller/ImagePostController.php
@@ -62,7 +62,7 @@ class ImagePostController extends AbstractController
 
         $message = new AddPonkaToImage($imagePost->getId());
         $envelope = new Envelope($message, [
-            new DelayStamp(5000)
+            new DelayStamp(500)
         ]);
         $messageBus->dispatch($envelope);
 
