diff --git a/config/messenger-worker.ini b/config/messenger-worker.ini
index 1a5c3f8..159dd8a 100644
--- a/config/messenger-worker.ini
+++ b/config/messenger-worker.ini
@@ -1,7 +1,7 @@
 [program:messenger-consume]
-command=php /path/to/your/app/bin/console messenger:consume async --time-limit=3600
-user=ubuntu
+command=php /Users/weaverryan/Sites/messenger/bin/console messenger:consume async --time-limit=3600
+user=weaverryan
 numprocs=2
 autostart=true
 autorestart=true
-process_name=%(program_name)s_%(process_num)02d
\ No newline at end of file
+process_name=%(program_name)s_%(process_num)02d
